### ISSUE TITLE
Constant used in attribute on local function warns that it is unused

### ISSUE THREAD

**Issue by PathogenDavid** [dotnet/roslyn#47619]
**Version Used**: Compiler version: '3.8.0-4.20460.4 (64f2392e)'. Language version: 9.0.

**Steps to Reproduce**:

Compile the following code: ([SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGAdAJQFcA7GASzDlwElTEB7ABwGVEA3MgYzigG4BYAFBCUAZiwAmDABUeMIQG8hGFVnEoALDLkBZODAAW9ACYAKAJTLVSwarsZO9YrCzZMxgDYeAchEoYAXgwAIllYXE8PYIFbexUrOIBtABEvajBGegQYU0jfSnMAXQT7HAA2DDgAD3gEYiwtX3I2OBQAVnInCxi4+Ni4prIW9s7ibpKAXyEJoA==))

```csharp
using System.Runtime.InteropServices;

public class Test
{
    public void TestMethod()
    {
        const string dllName = "Test.dll";
        
        [DllImport(dllName)]
        static extern void NativeFunction();
        
        NativeFunction();
    }
}
```

**Expected Behavior**: No warnings or errors.

**Actual Behavior**:

The compiler warns that `dllName` is not used:

```
warning CS0219: The variable 'dllName' is assigned but its value is never used
```

**Comment by RikkiGibson** [comment]
Is this a compiler bug @sharwell?
